### DISCUSSION TITLE
Dirtyfields doesn't get updated for inputs without ref

### DISCUSSION THREAD

**Discussion by adriancoliba** [react-hook-form/react-hook-form#5723] [Q&A] [answered]
Hello,

https://codesandbox.io/s/react-hook-form-watch-with-radio-buttons-and-select-examples-forked-3ibl3?file=/src/index.js
In the above example there are two inputs registered without a ref: 

```
  useEffect(() => {
    register("firstName", { required: true });
    register("lastName");
  }, [register]);
```
Unfortunately, dirtyFields remains {} if I touch/make changes to the inputs. 
If you're wondering why I'm not using `<input ref={register} />` instead, it is because I have a complicated dropdown in reality, consisting of a button and an unordered list: 
```
<div>
  <button/>
  <ul>
    <li></li>
    <li></li>
    <li></li>
  </ul>
</div>
```
Is there any way to update dirtyFields of these inputs without a ref? Thank you :)

EDIT: 
I found this example that uses a similar custom select (dropdown) component and console logging dirtyFields returns {}
https://codesandbox.io/s/custom-select-component-with-react-hook-form-forked-6thvo?file=/src/App.tsx

**Comment by bluebill1049** [comment, marked as answer]
`setValue('name', 'value', { shouldDirty: true })`

<img width="869" alt="Screen Shot 2021-06-25 at 9 17 34 am" src="https://user-images.githubusercontent.com/10513364/123343961-2e16f800-d596-11eb-8d76-946008131c2c.png">

https://react-hook-form.com/api/useform/setvalue

**Reply by adriancoliba** [reply]
How could I have missed this. Thank you so much bluebill1049

**Reply by bluebill1049** [reply]
Np.
